service/migration_manager.cc

```diff
@@ -1227,14 +1229,18 @@ future<> migration_manager::on_change(gms::inet_address endpoint, gms::applicati
             return make_ready_future();
         }
         if (_storage_proxy.get_token_metadata_ptr()->is_normal_token_owner(endpoint)) {
-            schedule_schema_pull(endpoint, *ep_state);
+            if (!_group0_client.using_raft()) {
+                schedule_schema_pull(endpoint, *ep_state);
+            }
```

Comment:
Why not put this inside schedule_schema_pull()? It's already "maybe" inside. For guaranteed pulls we do merge_schema_from().

Reply:
> Why not put this inside schedule_schema_pull()? It's already "maybe" inside. 

That's super confusing, considering there's also `maybe_schedule_schema_pull()`. And of course neither have a comment specifying their behavior.

IMO checking outside makes more sense.